cumulus/parachains/integration-tests/emulated/common/src/macros.rs

```diff
@@ -446,7 +449,23 @@ macro_rules! test_dry_run_transfer_across_pk_bridge {
 				// We assert the dry run succeeds and sends only one message to the local bridge hub.
 				assert!(result.execution_result.is_ok());
 				assert_eq!(result.forwarded_xcms.len(), 1);
-				assert_eq!(result.forwarded_xcms[0].0, VersionedLocation::from(Location::new(1, [Parachain($sender_bridge_hub::para_id().into())])));
+				let (destination, messages) = result.forwarded_xcms[0].clone();
+				assert_eq!(destination, VersionedLocation::from(Location::new(1, [Parachain($sender_bridge_hub::para_id().into())])));
+				assert_eq!(messages.len(), 1);
+				remote_message = messages[0].clone();
+			});
+
+			$sender_bridge_hub::execute_with(|| {
```

Comment:
@franciscoaguirre 
I think we should be able to test or dry-run the full route: AHK -> BHK ... BHP -> AHP.

This way, in the end, we'll be able to write or run a periodic TypeScript/PAPI/whatever script that tests/simulates/dry-runs an asset transfer from AHK to AHP live, without the need to manually intervene (no chopsticks required).


I think the only missing piece is the `DescendOrigin` I mentioned [here](https://github.com/paritytech/polkadot-sdk/issues/5708). We need to trigger some `pre-dispatch`, `fake-message-proof-delivery`, or whatever runtime API on the bridged BridgeHub, that will ultimately append the `DescendOrigin`. Let me think about it more.

Reply:
doesn't the dispatch before sending to outbound HRMP queue? I mean outbound HRMP Q of BH will contain full XCM exactly as it goes to AH, so I don't see the problem

Reply:
well, not exactly, bridged/target BridgeHub appends to the xcm `DescendOrigin` when dispatching: https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/xcm-builder/src/universal_exports.rs#L433-L439

so the XCM in outbound Q on sender BridgeHub is not exactly the same as it comes to the target AH

Reply:
but the appending `DescendOrigin` we can do "offchain" (in the test, PAPI call), so this PR should be enough probably

Reply:
I'm thinking of adding chopsticks tests for this under `integration-tests/chopsticks/` but in another PR